src/commonMain/kotlin/org/hisp/dhis/rules/api/RuleEngine.kt

```diff
@@ -1,15 +1,18 @@
 package org.hisp.dhis.rules.api
 
 import org.hisp.dhis.rules.engine.DefaultRuleEngine
+import org.hisp.dhis.rules.getEnvironment
 import org.hisp.dhis.rules.models.*
+import org.hisp.dhis.rules.models.TriggerEnvironment.SERVER
+import kotlin.jvm.JvmOverloads
```

Comment:
Well, after a second thought, I do think we need to annotate the methods as overloaded if we want to make it optional in Java. In Kotlin, the last parameter would be optional, but in Java it will be mandatory unless the methods are mark as overloaded

Reply:
Unfortunately, @JvmOverloads is not allowed on interfaces and default parameters are not allowed on overriding methods.
So I am manually creating the overload methods.
Another option would be not to use an interface.
I am merging this so I can test the artifact but then we can discuss the best solution